projects/js-packages/charts/src/components/pie-chart/pie-chart.tsx

```diff
@@ -243,21 +243,42 @@ const PieChartInternal = ( {
 										pathProps.onMouseLeave = onMouseLeave;
 									}
 
+									// Estimate text width more accurately for background sizing
+									const fontSize = 12;
+									const estimatedTextWidth = arc.data.label.length * fontSize * 0.6; // Rough estimate
+									const labelPadding = 6;
+									const backgroundWidth = estimatedTextWidth + labelPadding * 2;
+									const backgroundHeight = fontSize + labelPadding * 2;
+
 									return (
 										<g key={ `arc-${ index }` }>
 											<path { ...pathProps } />
 											{ hasSpaceForLabel && (
```

Comment:
We should probably skip the rendering if the label is set to empty

Reply:
<img width="869" height="1884" alt="image" src="https://github.com/user-attachments/assets/00af3145-2cc5-4ccd-adbe-fe89ad16c369" />